### DISCUSSION TITLE
icon in loading not appearing

### DISCUSSION THREAD

**Discussion by rpimentaf** [komarovalexander/ka-table#212] [Q&A]
Hi, @komarovalexander!!! 
In the Loading Example [https://komarovalexander.github.io/ka-table/#/loading] there's a icon with text when loading property is enabled. But, at my code, this icon never apperars, just the text. Do you know why this is happening?
Tks!

**Comment by komarovalexander** [comment]
Hi @rpimentaf  could you share the example? thanks!

**Comment by rpimentaf** [comment]
@komarovalexander , attached a picture of what i have when loading the table. See that the table is loading but the icon do not appear; only the text....
Tks!
![pic1](https://user-images.githubusercontent.com/69866458/185158677-85507c07-1d70-4c66-8afe-7dc0c49247c7.png)

**Comment by komarovalexander** [comment]
it is interesting because it is just an animation 

```
.ka-loading-icon{
  width: 10px;
  height: 10px;
  border-radius: 50%;
  animation: typing 1s linear infinite alternate;
  position: relative;
  left: -20px;
}
@keyframes typing{
  0%{
    background-color: $ka-loading-background-color-base;
    box-shadow: 20px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8),
                40px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8);
  }
  25%{
    background-color: transparentize($ka-loading-background-color-base, 0.6);
    box-shadow: 20px 0px 0px 0px $ka-loading-background-color-base,
                40px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8);
  }
  75%{
    background-color: transparentize($ka-loading-background-color-base, 0.6);
    box-shadow: 20px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8),
                40px 0px 0px 0px $ka-loading-background-color-base;
  }
  100%{
    box-shadow: none;
  }
}
```

which browser do you use? 
do you see the loading on that page https://komarovalexander.github.io/ka-table/#/loading? 
maybe you could reproduce it here https://stackblitz.com/edit/table-loading-ts?file=Demo.tsx and share the code with me?

**Comment by rpimentaf** [comment]
@komarovalexander , below the answer to your questions...
I'm using Chrome, version 104.0.5112.101.
Yes, I see the icon in that page.
And, finally, i've shared with the code i'm using.... 

Tks again!!!

**Reply by komarovalexander** [reply]
can you provide the link to the code? =)

**Reply by rpimentaf** [reply]
Sorry, man! T1m newbie with Stackblitz...

https://stackblitz.com/edit/react-ts-8hzru6?file=DadosVoos.tsx

**Reply by komarovalexander** [reply]
@rpimentaf do you see the loading indicator on this example? https://stackblitz.com/edit/react-ts-g2nxeo?file=DadosVoos.tsx,App.tsx (I took your code and removed your components)

**Reply by rpimentaf** [reply]
Yes, I do!
Strange... Any ideas?

**Comment by komarovalexander** [comment]
I do not see the code and styles of Page component but maybe it affects Table somehow,
can I ask you to move the table to the top level? 
```
    <>
      <Table {...tableProps} dispatch={dispatch} />
      <Page title="Dados da Consulta de Voos">...</Page>
    </>
```
do you see the Indicator in that case?

**Reply by rpimentaf** [reply]
no, @komarovalexander ! Unfortunely...

**Reply by rpimentaf** [reply]
@komarovalexander, is there anything else that i can do to help you find the possible reason to that behavior?

**Reply by komarovalexander** [reply]
Analyze style of the indicator element in chrome developer tools. By some reason styles could be overridden

**Comment by komarovalexander** [comment]
please check your code if you have another `@keyframes typing` in the styles. Maybe it overrides ka-table styles. I think I will rename @keyframes `typing` to `ka-loading-animation` in the next releases

**Comment by komarovalexander** [comment]
fix could be: 
```
.ka-loading-icon{
  animation: ka-loading-animation 1s linear infinite alternate;
}
@keyframes ka-loading-animation{
  0%{
    background-color: $ka-loading-background-color-base;
    box-shadow: 20px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8),
                40px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8);
  }
  25%{
    background-color: transparentize($ka-loading-background-color-base, 0.6);
    box-shadow: 20px 0px 0px 0px $ka-loading-background-color-base,
                40px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8);
  }
  75%{
    background-color: transparentize($ka-loading-background-color-base, 0.6);
    box-shadow: 20px 0px 0px 0px transparentize($ka-loading-background-color-base, 0.8),
                40px 0px 0px 0px $ka-loading-background-color-base;
  }
  100%{
    box-shadow: none;
  }
}
```
add it to your project styles after importing ka-table.css

**Reply by rpimentaf** [reply]
@komarovalexander, i'm using a another's man architeture because of regulations where i work. So, i'm not able to follow what you've recomended. Is it possible to change the files style.css and style.scss inside ka-table folder?

**Comment by komarovalexander** [comment]
Hi @rpimentaf 
Update the ka-table to v7.4.2 and check again please

**Reply by rpimentaf** [reply]
@komarovalexander , i think i have some limitations to upgrade. The latest version i'm able to install is 7.4.1... I tried forcing with add ka-table@7.4.2 but only 7.4.1 is available.
Another way to install it?

**Reply by komarovalexander** [reply]
I think this is because of some cache on your proxy. You can try to update it later

**Reply by komarovalexander** [reply]
Hi @rpimentaf did you manage to update the library version?

**Reply by rpimentaf** [reply]
yes, i did install the 7.4.2 version, but the icon still do not appear.... Tell me if you wanna more tests. Tks!!

**Reply by komarovalexander** [reply]
Can you open developer tools in chrome, click by .ka-loading-icon element and send me the screenshot of its css styles?

**Reply by rpimentaf** [reply]
I loaded the page that contais the ka-table, opened the  Chrome Tools. Choosed tab Elements (above) and styles tab (below) simultaneously.  And at tab styles, writed ".ka-loading-icon" in filter field. The result (styles) became empty. Is it correct?
![image](https://user-images.githubusercontent.com/69866458/185767139-01ec5b6c-aef4-4bc4-b13c-f356cbc4f360.png)

**Reply by komarovalexander** [reply]
Hi @rpimentaf you must search without dot, just “ka-loading-icon”

**Reply by rpimentaf** [reply]
Found it!!!
![image](https://user-images.githubusercontent.com/69866458/185797127-fbfe228b-3e30-46c3-ad35-24bbd114cbe9.png)

**Reply by komarovalexander** [reply]
Everything is alright and this should be shown, I have no idea what is the cause but it is something outside of the library. From my side I can only add an api for customization of loading panel to give ability to write custom html into that

**Reply by komarovalexander** [reply]
please check why those two options are crossed out 
it looks like this is the reason why animation is not working 
![image](https://user-images.githubusercontent.com/17903747/185802002-6215980d-b113-45fe-af05-0b9b798f27ca.png)

**Reply by komarovalexander** [reply]
do not filter by 'ka' in styles and you will see what overrides `animation-duration` and `animation-iteration-count` options

**Comment by rpimentaf** [comment]
Alexander, good morning!!!
I've tried what you've said, but the ka-loading icon still do not appear,
although the ka-table components wre present...
[image: image.png]
[]s



    Rodrigo Pimenta de Figueiredo
        ***@***.***


Em dom., 21 de ago. de 2022 às 03:02, Alexander Komarov <
***@***.***> escreveu:

**Reply by komarovalexander** [reply]
Hi, image was not attached..

**Reply by komarovalexander** [reply]
ka-loading Is shown according to initial screenshot, ka-loading-icon should be there as well because it shown unconditionally just try to search that when loading is shown 🙂

**Comment by komarovalexander** [comment]
Hi @rpimentaf  starting from 7.5.0 version you can customize loading panel content 
just add to the table component
```
 childComponents={{
          loading: {
            content: ({ text }) => (
              <>
                <img
                  height="100"
                  style={{ marginBottom: -40 }}
                  src="https://mir-s3-cdn-cf.behance.net/project_modules/disp/dae67631234507.564a1d230a290.gif"
                  alt="loading.."
                />
                <div className="ka-loading-text">{text}</div>
              </>
            ),
          },
        }}
```

see the demo: https://stackblitz.com/edit/table-loading-ts-yjbtrs?file=Demo.tsx
release info: https://github.com/komarovalexander/ka-table/releases/tag/7.5.0

**Reply by rpimentaf** [reply]
@komarovalexander ,  the icon appeared using this last solution you provided in 7.5.0 version. Great job and thanks very much and congratulations for your ka-table library!!!!
